### PR TITLE
Improve quality aa1

### LINKED ISSUE #3: Improve code quality
**Is your feature request related to a problem? Please describe.**
- Currently there is no `#region` and `#endregion` in the code.
- Currently we do not use prop types.

**Describe the solution you'd like**
- Implement `#region` and `#endregion` in every major component.
- Use prop types wherever possible.

### PR DESCRIPTION
**Pull request by Aichnerc**
**Checklist:**

 
 - [x] Have you added an explanation of what your changes do and why you'd like them to be included?
 - [ ] Have you updated or added documentation for the change?
 - [x] Have you tested your changes with successful results?

**Type of Changes:**

 
 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [x] New feature (non-breaking change which adds functionality)
 - [x] Documentation (non-breaking change which adds documentation)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change)

**What is the current behavior? (link to any open issues here)**
- Currently our `ScrollToTop` fix is not implemented.
- Currently we do not use `#regions`. #3  
- Currently there is no Prettier configuration.
- Currently only the main mdb react pro package is being excluded from github by our `.gitignore`.

**What is the new behavior (if this is a feature change)?**
- The `ScrollToTop` fix is implemented.
- We now use `#regions` on all files. The code has been cleaned up. Closes #3 
- A prettier configuration has been added.
- All possible future mdb pro packages are now being excluded from github by our `.gitignore`.

**Other information**:
- The version has been updated (minor).

### CODE REVIEW

**Review by LucaAllmaier — APPROVED** [review]
Looks good.

File: src/App.js [hunk 1]
```diff
@@ -9,32 +10,39 @@ import { BrowserRouter as Router } from "react-router-dom";
...
+
+//> Routes
 import Routes from "./Routes";
+//#endregion
 
+//#region > Components
 class App extends React.Component {
   render() {
     return (
       <Router>
```
Comment (schettn) [inline src/App.js:25]: I would recommend moving the Router to index.js. If you disagree leave it as it is.
https://github.com/snek-at/front/issues/54

**Review by schettn — APPROVED** [review]
Yep

File: src/App.js [hunk 2]
```diff
@@ -9,32 +10,39 @@ import { BrowserRouter as Router } from "react-router-dom";
...
  */
-import {
-  Footer,
-  Navbar,
-} from "./components/molecules";
-// Routes
+import { Footer, Navbar } from "./components/molecules";
+// Starts the page on top when reloaded or redirected
+import { ScrollToTop } from "./components/atoms";
+
```
Comment (pinterid) [inline src/App.js:16]:
```suggestion
```

File: src/App.js [hunk 3]
```diff
@@ -9,32 +10,39 @@ import { BrowserRouter as Router } from "react-router-dom";
...
+          </div>
+        </ScrollToTop>
       </Router>
     );
   }
 }
+//#endregion
 
+//#region > Exports
 export default App;
```
Comment (pinterid) [inline src/App.js:42]: In SNEK we added comments, to define the export. If this is to much overhead ignore it.
```suggestion
//#region > Exports
//> Default Class
export default App;
```

File: src/components/atoms/ScrollToTop/index.jsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+  render() {
+    return this.props.children;
+  }
+}
+//#endregion
+
+//#region > Exports
+//> Default Class
+export default withRouter(ScrollToTop);
```
Comment (pinterid) [inline src/components/atoms/ScrollToTop/index.jsx]: You are switching between coding guide lines in the files, either choice to write this comment for every export or remove it from every export.

File: src/components/molecules/Footer/index.jsx [hunk 5]
```diff
@@ -19,7 +20,9 @@ import "./footer.scss";
```
Comment (pinterid) [inline src/components/molecules/Footer/index.jsx:14]: This line and the line above "//> CSS" (cannot be marked) should be removed.
```suggestion
```

File: src/components/atoms/ScrollToTop/index.jsx (hunk 4 above)
Comment (schettn) [reply src/components/atoms/ScrollToTop/index.jsx]: Btw you are not exporting a default class here. It don't know exactly what withRouter() returns but it is not a class.

File: src/components/molecules/Navbar/index.jsx [hunk 6]
```diff
@@ -1,3 +1,4 @@
+//#region > Imports
 //> React
 // Contains all the functionality necessary to define React components
 import React from "react";
```
Comment (pinterid) [inline src/components/molecules/Navbar/index.jsx:4]: Remove the newline below this line. (cannot be marked)

File: src/components/pages/HomePage/index.jsx [hunk 7]
```diff
@@ -1,3 +1,4 @@
+//#region > Imports
 //> React
 // Contains all the functionality necessary to define React components
 import React from "react";
```
Comment (pinterid) [inline src/components/pages/HomePage/index.jsx:4]: Remove newline below this line. (Cannot be marked)

File: src/components/pages/HomePage/index.jsx [hunk 8]
```diff
@@ -28,7 +29,9 @@ import Projects from "../../../assets/content/projects.jpg";
```
Comment (pinterid) [inline src/components/pages/HomePage/index.jsx:29]:
```suggestion
```

File: src/components/pages/LoginPage/index.jsx [hunk 9]
```diff
@@ -1,7 +1,8 @@
+//#region > Imports
```
Comment (pinterid) [inline src/components/pages/LoginPage/index.jsx:1]: This import region is messed up, only add a newline between internal and external imports and place external imports above internal.

File: src/components/pages/MessagePage/index.jsx [hunk 10]
```diff
@@ -1,3 +1,4 @@
+//#region > Imports
 //> React
 // Contains all the functionality necessary to define React components
 import React from "react";
```
Comment (pinterid) [inline src/components/pages/MessagePage/index.jsx:4]: Remove newline below this line.

File: src/components/pages/ProfilePage/index.jsx [hunk 11]
```diff
@@ -1,3 +1,4 @@
+//#region > Imports
```
Comment (pinterid) [inline src/components/pages/ProfilePage/index.jsx:1]: This import region is messed up, only add a newline between internal and external imports and place external imports above internal.

File: src/components/pages/ProfilePage/index.jsx [hunk 12]
```diff
@@ -50,18 +50,17 @@ class ProfilePage extends React.Component {
...
         </MDBContainer>
       );
     }
   }
 }
+//#endregion
 
+//#region > Functions
 const mapStateToProps = (state) => {
   console.log(state);
```
Comment (pinterid) [inline src/components/pages/ProfilePage/index.jsx]: Is this still needed?

File: src/index.js [hunk 13]
```diff
@@ -1,3 +1,4 @@
+//#region > Imports
```
Comment (pinterid) [inline src/index.js:1]: This import region is messed up, only add a newline between internal and external imports and place external imports above internal.

**Review by pinterid — CHANGES_REQUESTED** [review]
Some changes have to be made.

File: src/App.js (hunk 2 above)
Comment (Aichnerc) [reply src/App.js:16]: We split imports per header and a new line.

File: src/App.js (hunk 3 above)
Comment (Aichnerc) [reply src/App.js:42]: Too much overhead, yes.

File: src/components/atoms/ScrollToTop/index.jsx (hunk 4 above)
Comment (Aichnerc) [reply src/components/atoms/ScrollToTop/index.jsx]: @schettn I copied this from `snek-at/front`. Please check if you find a better way to describe it there.

File: src/components/molecules/Footer/index.jsx (hunk 5 above)
Comment (Aichnerc) [reply src/components/molecules/Footer/index.jsx:14]: We split imports per header and a new line.

File: src/components/molecules/Navbar/index.jsx (hunk 6 above)
Comment (Aichnerc) [reply src/components/molecules/Navbar/index.jsx:4]: We split imports per header and a new line.

File: src/components/pages/HomePage/index.jsx (hunk 7 above)
Comment (Aichnerc) [reply src/components/pages/HomePage/index.jsx:4]: We split imports per header and a new line.

File: src/components/pages/HomePage/index.jsx (hunk 8 above)
Comment (Aichnerc) [reply src/components/pages/HomePage/index.jsx:29]: We split imports per header and a new line.

File: src/components/pages/LoginPage/index.jsx (hunk 9 above)
Comment (Aichnerc) [reply src/components/pages/LoginPage/index.jsx:1]: We split imports per header and a new line.

File: src/components/pages/MessagePage/index.jsx (hunk 10 above)
Comment (Aichnerc) [reply src/components/pages/MessagePage/index.jsx:4]: We split imports per header and a new line.

File: src/components/pages/ProfilePage/index.jsx (hunk 11 above)
Comment (Aichnerc) [reply src/components/pages/ProfilePage/index.jsx:1]: We split imports per header and a new line.

File: src/index.js (hunk 13 above)
Comment (Aichnerc) [reply src/index.js:1]: We split imports per header and a new line.

**Review by pinterid — APPROVED** [review]
Looks good.

**Pull request merged**